apiserver/facades/agent/secretsmanager/register.go

```diff
@@ -43,34 +43,34 @@ func NewSecretManagerAPIV1(context facade.Context) (*SecretsManagerAPIV1, error)
 	return &SecretsManagerAPIV1{SecretsManagerAPI: api}, nil
 }
 
+func checkPermission(context facade.Context) error {
+	if context.Auth().AuthUnitAgent() || context.Auth().AuthApplicationAgent() {
+		return nil
+	}
+	return context.Auth().HasPermission(permission.WriteAccess, names.NewModelTag(context.State().ModelUUID()))
```

Comment:
This doesn't belong here - this is an agent facade and should not be doing client facade permission checks.

Reply:
Ops, this was added for initial testing. Reverted!